### PR TITLE
Potential improvements for your RedBlackTree

### PR DESCRIPTION
**Pull request by ghost**
Your red black tree implementation is astonishing! My congrads on this! However i think i found some things that maybe need to change. Using #define NULL 0 seems not to be necessary and count variable must be of type size_t in order to handle even larger numbers. Also in the swap method you are using the same variable temp which type is int and using it to excange the value while the value is of type ElemType and not an integer

### CODE REVIEW

File: inc/RedBlackTree.h [hunk 1]
```diff
@@ -1,6 +1,6 @@
 #ifndef REDBLACKTREE_H
 #define REDBLACKTREE_H
-#define NULL 0
+// #define NULL 0
```
Comment (ezliu) [inline inc/RedBlackTree.h:3]: Let's delete this line instead of commenting it out.

**Comment by ezliu** [comment]
Thanks for your changes! They look great! I just want to take that line of code out instead of commenting it out.

**Comment by ghost** [comment]
Ok mate! Thanks for replying :+1:  you have done great job in this project

**Comment by ghost** [comment]
Oh and one more thing ! this is my first pull request so i don't know exactly what else i have to do on my part or all you have to do is approve the changes or not and thats all :P

**Comment by ezliu** [comment]
So normally, I think that you would make the modifications, commit, and push again, which would update this pull request, but I'm going to go ahead and change that one line -- thanks!

**Pull request merged**